### PR TITLE
New auth pattern

### PR DESCRIPTION
**Pull request by wolfovik**
Tested login and inventory.

### CODE REVIEW

**Comment by bukson** [comment]
I will test it and review today, thanks!

File: steampy/login.py [hunk 1]
```diff
@@ -47,26 +50,28 @@ def _send_login_request(self) -> Response:
...
             self.session.cookies.set(**store_cookie)
+
     @staticmethod
     def _create_cookie(name: str, cookie: str, domain: str) -> dict:
         return {"name": name,
                 "value": cookie,
                 "domain": domain}
 
     def _fetch_rsa_params(self, current_number_of_repetitions: int = 0) -> dict:
+        self.session.post(SteamUrl.COMMUNITY_URL)
```
Comment (bukson) [inline steampy/login.py:72]: why this post, it seems to be doing nothing

File: steampy/login.py [hunk 2]
```diff
@@ -47,26 +50,28 @@ def _send_login_request(self) -> Response:
...
             cookie = self.session.cookies.get_dict()[name]
-            community_cookie = self._create_cookie(name,cookie, community_domain)
-            store_cookie = self._create_cookie(name,cookie, store_domain)
+            community_cookie = self._create_cookie(name, cookie, community_domain)
+            store_cookie = self._create_cookie(name, cookie, store_domain)
             self.session.cookies.set(**community_cookie)
             self.session.cookies.set(**store_cookie)
+
     @staticmethod
     def _create_cookie(name: str, cookie: str, domain: str) -> dict:
```
Comment (bukson) [inline steampy/login.py:66]: move that to utils please

File: steampy/login.py [hunk 3]
```diff
@@ -117,11 +115,53 @@ def _assert_valid_credentials(login_response: Response) -> None:
...
-        for url in response_dict['transfer_urls']:
-            self.session.post(url, parameters)
+        for pass_data in parameters:
+            pass_data['params']['steamID'] = response_dict['steamID']
+            self.session.post(pass_data['url'], pass_data['params'])
 
     def _fetch_home_page(self, session: Session) -> Response:
         return session.post(SteamUrl.COMMUNITY_URL + '/my/home/')
+
+    def _update_steam_guard(self, login_response: Response) -> bool:
```
Comment (bukson) [inline steampy/login.py:128]: should return None as it just changes internal state, and we never use its value if it is ok

File: steampy/login.py [hunk 4]
```diff
@@ -117,11 +115,53 @@ def _assert_valid_credentials(login_response: Response) -> None:
...
+        }
+        response = self._api_call('POST', 'IAuthenticationService', 'UpdateAuthSessionWithSteamGuardCode',
+                                  params = update_data)
+        if response.status_code == 200:
+            self._pool_sessions_steam(client_id, request_id)
+            return True
+        else:
+            raise Exception('Cannot update steam guard')
+
+    def _pool_sessions_steam(self, client_id, request_id):
```
Comment (bukson) [inline steampy/login.py:149]: add types please

File: steampy/login.py [hunk 5]
```diff
@@ -117,11 +115,53 @@ def _assert_valid_credentials(login_response: Response) -> None:
...
+        pool_data = {
+            'client_id': client_id,
+            'request_id': request_id
+        }
+        response = self._api_call('POST', 'IAuthenticationService', 'PollAuthSessionStatus', params = pool_data)
+        self.refresh_token = response.json()["response"]["refresh_token"]
+
+    def _finallize_login(self):
```
Comment (bukson) [inline steampy/login.py:157]: add return type and fix typo, should be _finalize_login

File: steampy/login.py [hunk 6]
```diff
@@ -117,11 +115,53 @@ def _assert_valid_credentials(login_response: Response) -> None:
...
+        sessionid = self.session.cookies["sessionid"]
+        redir = "https://steamcommunity.com/login/home/?goto="
+
+        finallez_data = {
```
Comment (bukson) [inline steampy/login.py:161]: fix typo

File: steampy/login.py [hunk 7]
```diff
@@ -117,11 +115,53 @@ def _assert_valid_credentials(login_response: Response) -> None:
...
+        pool_data = {
+            'client_id': client_id,
+            'request_id': request_id
+        }
+        response = self._api_call('POST', 'IAuthenticationService', 'PollAuthSessionStatus', params = pool_data)
+        self.refresh_token = response.json()["response"]["refresh_token"]
+
+    def _finallize_login(self):
+        sessionid = self.session.cookies["sessionid"]
+        redir = "https://steamcommunity.com/login/home/?goto="
```
Comment (bukson) [inline steampy/login.py:159]: change variable name to be more descriptive and use COMMUNITY_URL

**Comment by bukson** [comment]
Please resolve the comments

Also if you can research the logout method, I was able to pass login test, but logout was unsucesfull.

I can merge this without logout working but it will be probably next complain.

File: steampy/login.py (hunk 2 above)
Comment (wolfovik) [reply steampy/login.py:66]: Done

File: steampy/login.py (hunk 3 above)
Comment (wolfovik) [reply steampy/login.py:128]: Done

File: steampy/login.py (hunk 5 above)
Comment (wolfovik) [reply steampy/login.py:157]: Done

File: steampy/login.py (hunk 7 above)
Comment (wolfovik) [reply steampy/login.py:159]: Done

File: steampy/login.py (hunk 6 above)
Comment (wolfovik) [reply steampy/login.py:161]: Done

**Comment by wolfovik** [comment]
> Please resolve the comments
> 
> Also if you can research the logout method, I was able to pass login test, but logout was unsucesfull.
> 
> I can merge this without logout working but it will be probably next complain.

I've never used the Logout function all these years.

**Review by bukson — APPROVED** [review]

**Pull request merged**